SubmittingPatches.rst

```diff
@@ -0,0 +1,442 @@
+
+     https://github.com/github/hub
+
+   allows you to submit pull requests directly from the command line::
+
+    $ hub pull-request -b ceph:master -h you:mything
+
+   Pull requests appear in the review queue at
+
+     https://github.com/organizations/ceph/dashboard/pulls
+
+   You may want to ping a developer in #ceph-devel on irc.oftc.net or on the
+   email list to ensure your submission is noticed.
+
+   When addressing review comments, can should either add additional patches to
+   your branch or (better yet) squash those changes into the relevant commits so
+   that the sequence of changes is "clean" and gets things right the first time.
+   The 'git rebase -i' command is very helpful in this process.  Once you have
+   updated your local branch, you can simply force-push to the existing branch
+   in your public repository that is referenced by the pull request with::
+
+     $ git push -f origin mything
+
+   and your changes will be visible from the existing pull-request.  You may want
+   to ping the reviewer again or comment on the pull request to ensure the updates
+   are noticed.
+
+   Q: Which branch should I target in my pull request?
+
+   A: The target branch depends on the nature of your change:
+
+      If you are adding a feature, target the "master" branch in your pull
+      request.
+
+      If you are fixing a bug, target the named branch corresponding to the
+      major version that is currently in development. For example, if
+      Infernalis is the latest stable release and Jewel is development, target
+      the "jewel" branch for bugfixes. The Ceph core developers will
+      periodically merge this named branch into "master". When this happens,
+      the master branch will contain your fix as well.
+
+      If you are fixing a bug (see above) *and* the bug exists in older stable
+      branches (for example, the "hammer" or "infernalis" branches), then you
+      should file a Redmine ticket describing your issue and fill out the
+      "Backport: <branchname>" form field. This will notify other developers that
+      your commit should be cherry-picked to these stable branches. For example,
+      you should set "Backport: hammer" in your Redmine ticket to indicate that
+      you are fixing a bug that exists on the "hammer" branch and that you
+      desire that your change be cherry-picked to that branch.
+
+   Q: How to include ``Reviewed-by: tag(s)`` in my pull request?
+
+      You don't. If someone reviews your pull request, they should indicate they
+      have done so by commenting on it with "+1", "looks good to me", "LGTM",
+      and/or the entire "Reviewed-by: ..." line with their name and email address.
+
+      The developer merging the pull request should note positive reviews and
+      include the appropriate Reviewed-by: lines in the merge commit.
+
+2. Patch submission via ceph-devel@vger.kernel.org
+
+   The best way to generate a patch for manual submission is to work from
```

Comment:
Should we add a small note here saying for the kernel client we want patches in mailing list and for everything else it is preferred to submit in Github for review?